### PR TITLE
Feat/update to new store

### PR DESCRIPTION
**Pull request by cdrx**

### CODE REVIEW

File: {{cookiecutter.package_name}}/.flake8 [hunk 1]
```diff
@@ -0,0 +1,26 @@
+[flake8]
+max-complexity = 15
+
+exclude = .git,
+    __pycache__,
+    packaging,
+    src/core/imm/core/migrations/*,
+    src/core/imm/core/closuretree/*,
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/.flake8:8]: packaging and src/core don't exist here, so they can be removed

File: {{cookiecutter.package_name}}/.pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+exclude: '^$|^frontend/|^src/frontend|^vendor/'
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/.pre-commit-config.yaml:1]:
```suggestion
exclude: '^$'
```

File: {{cookiecutter.package_name}}/.pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.7.4
+    hooks:
+      - id: pyupgrade
+        args: [--py38-plus]
+  - repo: https://github.com/myint/autoflake
+    rev: v1.4
+    hooks:
+      - id: autoflake
+        args: ['--in-place', '--imports=django,requests,rest_framework', '--remove-duplicate-keys']
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/.pre-commit-config.yaml:38]:
```suggestion
        args: ['--in-place', '--imports=django,requests,rest_framework,imm', '--remove-duplicate-keys']
```

File: {{cookiecutter.package_name}}/.pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      - id: isort
+  - repo: https://github.com/psf/black
+    rev: 20.8b0
+    hooks:
+      - id: black
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 3.8.4
+    hooks:
+      - id: flake8
+        exclude: '^src/imm/migrations/|^src/imm/closuretree/'
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/.pre-commit-config.yaml:68]: This line can be removed

File: {{cookiecutter.package_name}}/.github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+#name: Build and upload extension
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/.github/workflows/release.yml:1]: Needs updating with the copy from the test extension

File: {{cookiecutter.package_name}}/Makefile [hunk 6]
```diff
@@ -1,32 +1,63 @@
-bold := $(shell tput bold)
-byel := $(bold)$(shell tput setaf 11)
-end := $(shell tput sgr0)
 # extracts the package name from imm-extension.location in pyproject.toml
 PACKAGE = $(shell sed -n 's/.*location *= "*\([^ ]*.*\)/\1/p' ./pyproject.toml | cut -d. -f1)
+EXTENSION = $(shell sed -n 's/.*extension *= "*\([^ ]*.*\)"/\1/p' ./pyproject.toml | cut -d. -f1)
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/Makefile:3]: Needs updating with the copy from the test extension

File: {{cookiecutter.package_name}}/pyproject.toml [hunk 7]
```diff
@@ -1,20 +1,7 @@
...
-target-version = ['py38']
-
-
-[tool.cruft]
-skip = ["{{cookiecutter.package_name}}", "README.md", "CHANGELOG.md"]
-
-
 [imm-extension]
 name = "{{cookiecutter.package_name|replace('_', ' ')|title}}"
 version = "master"
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/pyproject.toml:3]: Semantic will update the version number here now, so we should start it at "1.0.0"

File: {{cookiecutter.package_name}}/pyproject.toml [hunk 8]
```diff
@@ -28,3 +15,33 @@ include-files = [
     "templates/**/*.html",
     "static/**/*",
 ]
+
+[build-system]
+requires = ["setuptools", "wheel", "cython", "toml"]
+
+
+[tool.black]
+line-length = 120
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/pyproject.toml:24]:
```suggestion
line-length = 125
```

File: {{cookiecutter.package_name}}/pyproject.toml [hunk 9]
```diff
@@ -28,3 +15,33 @@ include-files = [
...
+skip-string-normalization = true
+target-version = ['py38']
+
+
+[tool.cruft]
+skip = ["{{cookiecutter.package_name}}", "README.md", "CHANGELOG.md"]
+
+[tool.isort]
+profile = "black"
+known_first_party = "imm_test"
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/pyproject.toml:34]:
```suggestion
known_first_party = "{{cookiecutter.package_name}}"
```

**Comment by cdrx** [comment]
Can you update this template to use pytest also? We'll need to:

- Move the tests folder into the root, rather than next to `extension.py`
- Replace the unit test stuff with a basic pytest (copy from the test repo maybe)

File: {{cookiecutter.package_name}}/.pre-commit-config.yaml (hunk 4 above)
Comment (Pete-Hamlin) [reply {{cookiecutter.package_name}}/.pre-commit-config.yaml:68]: Just to clarify just `line 68`?

Comment (cdrx) [reply {{cookiecutter.package_name}}/.pre-commit-config.yaml:68]: Yeh the excludes

File: {{cookiecutter.package_name}}/tests/test_extension.py [hunk 10]
```diff
@@ -1,10 +1,7 @@
 from unittest import TestCase
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/tests/test_extension.py:1]:
```suggestion

```

File: {{cookiecutter.package_name}}/tests/extension_test.py [hunk 11]
```diff
@@ -0,0 +1,7 @@
+from unittest import TestCase
+
```
Comment (cdrx) [inline {{cookiecutter.package_name}}/tests/extension_test.py:2]:
```suggestion
```

**Pull request merged**